app/controllers/TicketBooking.scala

```diff
@@ -0,0 +1,204 @@
+
+  private val availableScreenings: List[Screening] = DefaultScreenings.getAvailableScreenings
+  private var availableScreeningsFiltered: List[Screening] = Nil
+  private var chosenScreeningOption: Option[Screening] = None
+  private var chosenSeats: Seq[(Int, Int)] = Nil
+  private var chosenTickets: Seq[Ticket] = Nil
+
+  val customerDataForm: Form[Customer] = Form(mapping(
+    "name" -> text(3),
+    "surname" -> text(3)
+  )(Customer.apply)(Customer.unapply))
+
+  def ticketBookingInit: Action[AnyContent] = Action { implicit request =>
+    val csrfToken = CSRF.getToken.get.value
+    Ok(views.html.ticketBookingInit()).withSession("csrfToken" -> csrfToken)
+  }
+
+  def chooseDateAndTime: Action[AnyContent] = Action { implicit request =>
+    val postVals = request.body.asFormUrlEncoded
+    postVals.map { args =>
+      if (args("dateTime").head.nonEmpty) {
+        val chosenDateTime = args("dateTime").head
+        println(chosenDateTime)
+        Redirect(routes.TicketBooking.availableMovies).withSession(request.session.+("dateTime" -> chosenDateTime))
+      } else {
+        println("empty date time")
+        Redirect(routes.TicketBooking.ticketBookingInit).flashing("error" -> "choose date and time!")
+      }
+    }.getOrElse(Redirect(routes.TicketBooking.ticketBookingInit))
+  }
+
+  def availableMovies: Action[AnyContent] = Action { implicit request =>
+    val chosenDateTimeTry = Try(LocalDateTime.parse(request.session.get("dateTime").get))
+    chosenDateTimeTry match {
+      case Failure(_) =>
+        Redirect(routes.TicketBooking.ticketBookingInit)
+          .withNewSession.flashing("error" -> "wrong date-time format")
+      case Success(dateTime) =>
+        availableScreeningsFiltered =
+          availableScreenings.filter(screening =>
+            screening.startTime.plusMinutes(90).isAfter(dateTime) &&
+              screening.startTime.minusHours(3).isBefore(dateTime) &&
+              screening.startTime.minusMinutes(15).isAfter(LocalDateTime.now())
+          )
+        Ok(views.html.availableMovies(availableScreeningsFiltered)).withSession(request.session)
+    }
+  }
+
+  def chooseScreening: Action[AnyContent] = Action { implicit request =>
+    val postVals = request.body.asFormUrlEncoded
+    postVals.map { args =>
+      val title = args("title").head
+      val startTime = args("startTime").head
+      chosenScreeningOption = availableScreenings
+        .find(scr => scr.movie.title == title && scr.startTime == LocalDateTime.parse(startTime))
+      chosenScreeningOption match {
+        case None => Redirect(routes.TicketBooking.availableMovies).withSession(request.session)
+        case Some(_) => Redirect(routes.TicketBooking.availableSeats).withSession(request.session)
+      }
+    }.getOrElse(Redirect(routes.TicketBooking.ticketBookingInit).withNewSession)
+  }
+
+  def availableSeats: Action[AnyContent] = Action { implicit request =>
+    val screeningTry = Try(chosenScreeningOption.get)
```

Comment:
why did you decide to use `Try` here?

Reply:
chosenScreeningOption is a var that is declared at the beginning of a session with:
private var chosenScreeningOption: Option[Screening] = None
Later, it is changed into Some(Screening) if the customer chose a screening properly before going to the "availableSeats" endpoint. If he did not choose the screening properly - for example went straight to the "availableSeats" endpoint without passing earlier endpoints properly - this var is still a None. So - in such a situation - I don't want my program to fail with an error or exception - and using Try helps to rescue the program from failure and also redirects the user properly.